src/App.tsx

```diff
@@ -13,43 +13,49 @@ import styled from 'styled-components'
 import Share from './pages/Share'
 
 export default function App() {
-  const { settings, update, togglePresence } = useSettings()
+  const {
+    settings: { students, absentStudents, rooms, groups, seed },
```

Comment:
This is interesting. Are you destructuring `settings` as well? I didn't know that was possible!

Reply:
Exactly. I usually try to stay away from it, as it looks really complicated, but here it helps.